pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -201,26 +203,41 @@ func (c *Cluster) setLoginCredentials(password string) error {
 	// Set the login credentials. Write the command/args to the debug log so we don't write the password by default to the log.
 	logger.Infof("setting ceph dashboard %q login creds", dashboardUsername)
 
+	// Generate a temp file
+	file, err := ioutil.TempFile("", "")
+	if err != nil {
+		return errors.Wrap(err, "failed to generate temp file")
+	}
+	err = ioutil.WriteFile(file.Name(), []byte(password), 0444)
+	if err != nil {
+		return errors.Wrap(err, "failed to write into file")
+	}
 	// retry a few times in the case that the mgr module is not ready to accept commands
-	_, err := client.ExecuteCephCommandWithRetry(func() (string, []byte, error) {
-		args := []string{"dashboard", "set-login-credentials", dashboardUsername, password}
+	_, err = client.ExecuteCephCommandWithRetry(func() (string, []byte, error) {
+		args := []string{"dashboard", "ac-user-create", dashboardUsername, "-i", file.Name(), "administrator"}
```

Comment:
At the next reconcile (e.g. If you restart the operator) does it skip setting the password?

Reply:
```
op-mgr: the dashboard secret was already generated
2021-04-08 05:15:41.178628 I | op-mgr: setting ceph dashboard "admin" login creds
2021-04-08 05:15:41.491418 I | op-mgr: successfully set ceph dashboard creds
2021-04-08 05:15:43.758269 I | op-mgr: successful modules: dashboard
```
looking to this I believe it is skipping but I'll confirm again today.

Reply:
actually, I was wrong, it creates passwords but skips creating self-signed certs.